### DISCUSSION TITLE
Response body replaced with error body when using RequireAny and OIDCUnAuthzAction

### DISCUSSION THREAD

**Discussion by takesson** [OpenIDC/mod_auth_openidc#934] [Q&A]
I had these Require statements at one point and saw some strange behaviour.
```
<Location /something >
  <RequireAll>
    Require valid-user
    Require method GET
    <RequireAny>
      Require claim roles:CmsAdmin
      Require claim roles:CmsUser
    </RequireAny>
  </RequireAll>
</Location>
```

The response was 200 OK but the body was replaces with:
```
<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01//EN" "http://www.w3.org/TR/html4/strict.dtd">
<html>
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8">
    <title>Authorization Error Redirect</title>
    <meta http-equiv="refresh" content="0; url=/openidc/errors/roles/">
  </head>
  <body>
<p></p>
  </body>
</html>
```

Seems to happen in combination with:
```
OIDCUnAutzAction 302 /openidc/errors/roles/
```

It can be worked around in this case by replacing RequireAny-section with a single Require claim regex.

**Comment by zandbelt** [comment]
thanks for the heads up, this limitation was indeed not documented; I have added a note [here](https://github.com/zmartzone/mod_auth_openidc/commit/9a04dba9fab14270548e2c458d4f9a9ea3063515) and to the wiki https://github.com/zmartzone/mod_auth_openidc/wiki/Step-up-Authentication

**Reply by takesson** [reply]
Good, do you think it is incompatible with RequireAll? 

Thanks

**Reply by zandbelt** [reply]
edit: it is incompatible with `RequireAny` and multiple Require statements since the module is only called for every Require statement individually but is unaware of the overall result, which may involve other authentication methods/modules as well

as it happens, it does work for `RequireAll` since the overall result is determined by the invididual Require statement as evaluated by the module, I'l adapt the docs for that
